### PR TITLE
MNT Update unit test

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/.github/issues/34

Fixes https://github.com/silverstripe/cwp-core/actions/runs/4559370105/jobs/8043562334#step:12:69

` 1) CWP\Core\Tests\PasswordStrengthTest::testPasswordMinLength
Error: Call to a member function getMinLength() on null`

Issue came about because of this PR https://github.com/silverstripe/recipe-core/pull/86.

I think there's a weird code-path for how the PasswordValidator normally gets registered in the CMS which I wasn't able to quickly find via a debug, though I've manually confirmed again that it does work so no issue there.  It's just whatever code path that is taken in the CMS isn't also taken by unit-tests

### CODE REVIEW

**Comment by GuySartorelli** [comment]
I don't feel comfortable merging this without understanding _why_ the thing wasn't being injected for this test - as it may indicate there's some edge case where it won't be available for projects as well.

**Comment by emteknetnz** [comment]
I think it wasn't being injected because of the code that was in recipe-core that got added to app/_config.php on `composer create-project`

File: tests/PasswordStrengthTest.php [hunk 1]
```diff
@@ -26,6 +28,12 @@
  */
 class PasswordStrengthTest extends SapphireTest
 {
+    protected function setUp(): void
+    {
+        parent::setUp();
+        Injector::inst()->registerService(new PasswordValidator(), PasswordValidator::class);
```
Comment (maxime-rainville) [inline tests/PasswordStrengthTest.php]: I think the problem is coming from this line in SapphireTest: https://github.com/silverstripe/silverstripe-framework/blob/c1427ff9c4c7711336eebfbbbf0f48bded509507/src/Dev/SapphireTest.php#L283-L285

When I comment out the line locally, the test passes without issue.

Not sure why SapphireTest wants to nerf the password validator ... my guess is that it gets in the way of some other testing logic. In this context, it seems reasonable to manually re-register the validator.

I think this approach is better because it will use the original validator rather than instantiating a new one.

```suggestion
        // Out of the box, SapphireTest wants to deregister the PasswordValidator. However, we need for these test to work.
        $validator = Member::password_validator();
        parent::setUp();
        Member::set_password_validator($validator);
```

**Review by maxime-rainville — CHANGES_REQUESTED** [review]

File: tests/PasswordStrengthTest.php (hunk 1 above)
Comment (emteknetnz) [reply tests/PasswordStrengthTest.php]: Updated

**Review by maxime-rainville — APPROVED** [review]
